### ISSUE TITLE
[Serious] Keyboard focus is lost or misplaced due to user interaction or content update

### ISSUE THREAD

**Issue by ntallman** [APTrust/dart#607]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87749 |
| **Checkpoint** | 2.4 Navigable › Focus Order (2.4.3.a) |
| **Impact** | Serious |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 02 Import/Export Workflow |
| **URL** | <https://aptrust.org/documentation-page/dart-digital-archivists-resource-tool/> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Date Reported** | 05/12/26 at 12:13 pm |
| **Issue Status** | Open |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | kim.casey@ablr360.com |

## Description

Keyboard focus is lost or placed on the wrong element during user interaction, content refresh or update, or other reason.
When you Import Settings (from JSON), an Import Result screen loads, but the focus goes to the top and the menu is announced instead of the Import Result.  See Import Export User Experience video at 1:25.

## Recommendation

Rule : 
The focus MUST be purposely moved or set (via JavaScript) onto the appropriate element when the user's action requires a change of context or location for effective keyboard or touch interaction.

How To Fix: 
Fix this issue by explicitly placing focus on a logical element when content is removed, refreshed, or added, for example:
1. For content added to the screen in reaction to a user-fired event, focus should be shifted to the new content - such as in single page applications.
2. For content removed from the screen in reaction to a user-fired event, focus should be shifted to the next logical place in the interaction.

Background: 
One of the biggest challenges when creating rich web interfaces using JavaScript is the management of focus when new content or controls are added or removed from the page. The presentation or modification of content on the screen may require that the user interact with -or at the very least, take notice of - the new or changed content. It is important to have a clear indication of page content that has been updated. This allows the user to understand the change and also allows a keyboard user to interact with any new content.

## Screenshots

![02_focus logical](https://raw.githubusercontent.com/APTrust/dart/master/docs/accessibility-audit/screenshots/87749/02_focus_logical.png)
